bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java

```diff
@@ -0,0 +1,243 @@
+            s.execute(evalContext);
+        } catch (ScriptExecutionException | ScriptParsingException e) {
+            throw new ScriptException(e.getMessage(), modelName, -1);
+        }
+        return null;
+    }
+
+    private DefaultEvaluationContext createEvaluationContext(IEvaluationContext specificContext) {
+        DefaultEvaluationContext evalContext = new DefaultEvaluationContext(specificContext);
+        for (Map.Entry<String, String> entry : implicitVars.entrySet()) {
+            Object value = context.getAttribute(entry.getKey());
+            if (value != null) {
+                evalContext.newValue(QualifiedName.create(entry.getValue()), value);
+            }
+        }
+        return evalContext;
+    }
+
+    @Override
+    public Object eval(Reader reader) throws ScriptException {
+        return null;
+    }
+
+    @Override
+    public Object eval(String script, Bindings n) throws ScriptException {
+        return null;
+    }
+
+    @Override
+    public Object eval(Reader reader, Bindings n) throws ScriptException {
+        return null;
+    }
+
+    @Override
+    public void put(String key, Object value) {
+
+    }
+
+    @Override
+    public Object get(String key) {
+        return null;
+    }
+
+    @Override
+    public Bindings getBindings(int scope) {
+        return null;
+    }
+
+    @Override
+    public void setBindings(Bindings bindings, int scope) {
+    }
+
+    @Override
+    public Bindings createBindings() {
+        return null;
+    }
+
+    @Override
+    public ScriptContext getContext() {
+        return context;
+    }
+
+    @Override
+    public void setContext(ScriptContext context) {
+
+    }
+
+    @Override
+    public ScriptEngineFactory getFactory() {
+        return new ScriptEngineFactory() {
+
+            @Override
+            public ScriptEngine getScriptEngine() {
+                return null;
+            }
+
+            @Override
+            public String getProgram(String... statements) {
+                return null;
+            }
+
+            @Override
+            public Object getParameter(String key) {
+                return null;
+            }
+
+            @Override
+            public String getOutputStatement(String toDisplay) {
+                return null;
+            }
+
+            @Override
+            public List<String> getNames() {
+                return null;
+            }
+
+            @Override
+            public List<String> getMimeTypes() {
+                return Collections.singletonList(MIMETYPE_OPENHAB_DSL_RULE);
+            }
+
+            @Override
+            public String getMethodCallSyntax(String obj, String m, String... args) {
+                return null;
+            }
+
+            @Override
+            public String getLanguageVersion() {
+                return "v1";
+            }
+
+            @Override
+            public String getLanguageName() {
+                return "openHAB Rule DSL";
```

Comment:
By the way @kaikreuzer, have you noticed the "o" gets capitalized between here and the UI? I don't think it's done client-side, so not sure where it happens.

Reply:
Yes, I had noticed this as well. I assume it must be the javax.script implementation that capitalizes it. I've now simply changed the name to "Rule DSL", so we don't have to bother anymore ;-)